### PR TITLE
Fix powershell escaping

### PR DESCRIPTION
**Pull request by rashil2000**
Fix what was left in #2641

### CODE REVIEW

**Comment by vovcacik** [comment]
@janlazo Thanks! That explains the implementation of `s:shellesc_ps1`. [^1] Fzf is not subject to the rules specified in *CommandLineToArgvW*, so it looks like internal requirement of the plugin. Correct me if I am wrong please.

Is there any scenario that would cause issues when the `quoteEntry` method wouldn't escape `"` with backslash. Note that `quoteEntry` escapes strings only when expanding fzf's placeholders (like `{}`, `{q}` etc). There is a gap in my understanding that would connect something like `fzf --bind "enter:execute(nvim {})` and the `jobstart` command from vim/nvim.

I am asking for this because @stinos is proposing to use powershell's native quoting rules and if that would not break any workflows I think @junegunn could actually like it more (the rules are closer to how fzf escapes strings on *nix).

[^1]: I am comparing the two `if` branches in [L1422](https://github.com/junegunn/vim-plug/blob/c9971346bb486ccff354aaee8606cc0d2d5a0c97/plug.vim#L1422) and [L1432](https://github.com/junegunn/vim-plug/blob/c9971346bb486ccff354aaee8606cc0d2d5a0c97/plug.vim#L1432)

**Comment by janlazo** [comment]
Any Windows program that executes external commands is subject to CommandLineToArgW.
golang does it by default.
golang doesn't special-case cmd.exe so `cmdline` is non-empty when the shell is cmd.exe.

Command passed to jobstart must be escaped. Editor plugin doesn't have to escape commands that fzf (binary) will execute internally. When executing the expanded command,  inner `"` must be escaped (on Windows).

Backtick escaping seems to specific to powershell. cmd.exe doesn't special-case backticks. It can work for the value passed to `-Command`.  I wouldn't use it when dealing with external programs.

**Comment by vovcacik** [comment]
New test suite added, with docs. It drops one level of escaping in favor of using powershell's native escaping. Good way to see the difference is to compare those two commits: [old][] and [new][].

[old]: https://github.com/vovcacik/fzf/commit/24df337ca970cabd355236255b72326fb85cebfd
[new]: https://github.com/vovcacik/fzf/commit/3f68fe3e0ed31640830411c664b3f2a60875de89

File: src/terminal_windows.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func quoteEntry(entry string) string {
 			return "^" + match
 		})
 	} else if strings.Contains(shell, "pwsh") || strings.Contains(shell, "powershell") {
-		escaped := strings.Replace(entry, `"`, `""`, -1)
+		escaped := strings.Replace(entry, `"`, `\"`, -1)
```
[the remark this answers is not in the document]
Comment (vovcacik) [reply src/terminal_windows.go:40]: @stinos, @rashil2000 the current implementation is actually what you guys want. The consistency with ripgrep (from cmd) would require additional level of escaping (as pointed out in  https://github.com/junegunn/fzf/pull/2647#issuecomment-955752297 and https://github.com/vovcacik/fzf/commit/5e774bc9d0d49713ab6aefeb9ffb0a15dc78fd12)

**Comment by junegunn** [comment]
@vovcacik Thanks for the test cases!
@rashil2000 Can you merge @vovcacik's PR to your branch? Let's wrap this up.

**Comment by junegunn** [comment]
Rebased to tidy up the commits.

**Comment by junegunn** [comment]
Merged, thanks!

**Pull request merged**